src/Uno.UI/Microsoft/UI/Xaml/Controls/RatingControl/RatingControl.cs

```diff
@@ -0,0 +1,1195 @@
+
+			UpdateRatingItemsAppearance();
+		}
+
+		private void OnMaxRatingChanged(DependencyPropertyChangedEventArgs args)
+		{
+			StampOutRatingItems();
+		}
+
+		private void OnPlaceholderValueChanged(DependencyPropertyChangedEventArgs args)
+		{
+			UpdateRatingItemsAppearance();
+		}
+
+		private void OnValueChanged(DependencyPropertyChangedEventArgs args)
+		{
+			// Fire property change for UIA
+			AutomationPeer peer = FrameworkElementAutomationPeer.FromElement(this);
+			if (peer != null)
+			{
+				var ratingPeer = peer as RatingControlAutomationPeer;
+				ratingPeer.RaisePropertyChangedEvent(Value);
+			}
+
+			UpdateRatingItemsAppearance();
+		}
+
+		private void OnIsEnabledChanged(object sender, DependencyPropertyChangedEventArgs args)
+		{
+			// MSFT 11521414 TODO: change states (add a state)
+			UpdateRatingItemsAppearance();
+		}
+
+		private void OnCaptionSizeChanged(object sender, SizeChangedEventArgs args)
+		{
+			ResetControlWidth();
+		}
+
+		private void OnPointerCancelledBackgroundStackPanel(object sender, PointerRoutedEventArgs args)
+		{
+			PointerExitedImpl(args);
+		}
+
+		private void OnPointerCaptureLostBackgroundStackPanel(object sender, PointerRoutedEventArgs args)
+		{
+			// We capture the pointer because we want to support the drag off the
+			// left side to clear the rating scenario. However, this means that
+			// when we simply click to set values - we get here, but we don't want
+			// to reset the scaling on the stars underneath the pointer.
+			PointerExitedImpl(args, false /* resetScaleAnimation */);
+		}
+
+		private void OnPointerMovedOverBackgroundStackPanel(object sender, PointerRoutedEventArgs args)
+		{
+			if (!IsReadOnly)
+			{
+				var point = args.GetCurrentPoint(m_backgroundStackPanel);
+				float xPosition = (float)point.Position.X;
+				if (SharedHelpers.IsAnimationsEnabled())
+				{
+					m_sharedPointerPropertySet.InsertScalar("starsScaleFocalPoint", xPosition);
+					var deviceType = args.Pointer.PointerDeviceType;
+
+					switch (deviceType)
+					{
+						case PointerDeviceType.Touch:
+							m_sharedPointerPropertySet.InsertScalar("pointerScalar", c_touchOverScale);
+							break;
+						default: // mouse, TODO: distinguish pen later
+							m_sharedPointerPropertySet.InsertScalar("pointerScalar", c_mouseOverScale);
+							break;
+					}
+				}
+
+				m_mousePercentage = (double)(xPosition) / CalculateActualRatingWidth();
+
+				UpdateRatingItemsAppearance();
+				args.Handled = true;
+			}
+		}
+
+		private void OnPointerEnteredBackgroundStackPanel(object sender, PointerRoutedEventArgs args)
+		{
+			if (!IsReadOnly)
+			{
+				m_isPointerOver = true;
+				args.Handled = true;
+			}
+		}
+
+		private void OnPointerExitedBackgroundStackPanel(object sender, PointerRoutedEventArgs args)
+		{
+			PointerExitedImpl(args);
+		}
+
+		private void PointerExitedImpl(PointerRoutedEventArgs args, bool resetScaleAnimation = true)
+		{
+			var point = args.GetCurrentPoint(m_backgroundStackPanel);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'point'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)